spec/models/manageiq/providers/vmware/infra_manager/event_catcher/runner_spec.rb

```diff
@@ -0,0 +1,115 @@
+require "spec_helper"
+
+describe ManageIQ::Providers::Vmware::InfraManager::EventCatcher::Runner do
+  let(:ems)      { FactoryGirl.create(:ems_vmware, :hostname => "hostname") }
+  let(:catcher)  { ManageIQ::Providers::Vmware::InfraManager::EventCatcher::Runner.new(:ems_id => ems.id) }
+  let(:settings) { {:flooding_monitor_enabled => false} }
+
+  before do
+    allow_any_instance_of(ManageIQ::Providers::Vmware::InfraManager).to receive(:authentication_check).and_return([true, ""])
+    allow_any_instance_of(MiqWorker::Runner).to receive(:worker_initialization)
+    allow_any_instance_of(MiqWorker::Runner).to receive(:worker_settings).and_return(settings)
```

Comment:
❤️   Yes, thank you!  rspec-mocks' `any_instance_of` should be the last choice when nothing else works.  Using it can easily break with each ruby version upgrade.